### PR TITLE
Pass optional query object to validateDomainContactInformation (2)

### PR DESCRIPTION
**Pull request by nbloomf**
#### Changes proposed in this Pull Request

* Companion to backend change D41243-code, although this patch is independent. Exposes a new optional parameter to the domain contact validation endpoint, allowing callers to pass in a `query` object to be passed on to `wpcom.req.post`. The immediate reason for this is to allow passing an `apiVersion` parameter.

To do:
* [x] Fix order of arguments (callback last)
* [x] ~Update consumers of this code to the new argument order~ (obsolete)

#### Testing instructions

This touches three pages:
* Normal domain contact form: enter checkout with a domain, make sure you can validate your contact info.
* Canadian domain contact form: enter checkout with a .ca domain, make sure you can validate your contact info.
* Domain contact update form: Try to update your contact information on an existing domain, make sure it can validate.

In all three cases I think it is enough to watch the network tab for requests to `me/domain-contact-information/validate` and check that the request and response look ok.

### CODE REVIEW

File: client/lib/wpcom-undocumented/lib/undocumented.js
Comment (sirbrillig) [inline client/lib/wpcom-undocumented/lib/undocumented.js:679]: This works great! Still I have a suggestion: what if instead of adding a boolean in the middle of the argument list, we added an options object at the end of the argument list. Since arguments are always optional in JS, that would mean that existing calls to the function would not need to change (the options object would simply be undefined in those cases). New calls to the function could add `, { qualifyProperties: true } )` at the end of their calls, which also has the benefit of being more clear when reading than `, true )`. What do you think?

Comment (sirbrillig) [reply client/lib/wpcom-undocumented/lib/undocumented.js:679]: Alternatively, we could change all the function arguments to properties of an object; that way they would be named and the order (or their absence) wouldn't matter. This would be more invasive, but equally useful.

So instead of

```js
function(
	contactInformation,
	domainNames,
	qualifyProperties,
        fn
)```

it would be

```js
function( {
	contactInformation,
	domainNames,
	qualifyProperties,
        fn
} )```

Comment (nbloomf) [reply client/lib/wpcom-undocumented/lib/undocumented.js:679]: An earlier version of this patch did add the parameter at the end, and I changed it because it felt unidiomatic for some reason. I do prefer the minimal nature of that, though, and using an object is a good idea. I'll make that edit.

Comment (sirbrillig) [reply client/lib/wpcom-undocumented/lib/undocumented.js]: I agree about the idiomatic point; callbacks feel "right" at the end... but the challenge is that if we go down that road then either the callback should be converted to a Promise or some or all of the arguments should be put in an object, and this is the most minimal, but elegant, change I could think of.

**Review by sirbrillig — APPROVED** [review]
👍

**Comment by nbloomf** [comment]
Thanks! I will hold off on merging this until and unless we get Cobalt's approval on D41243-code, since without that this change is useless.

**Review by sirbrillig — APPROVED** [review]
The validate endpoint looks like it's working correctly.

**Pull request merged**
